src/garage/sampler/off_policy_vectorized_sampler.py

```diff
@@ -186,28 +187,30 @@ def obtain_samples(self, itr, batch_size=None, whole_paths=True):
                             ['undiscounted_return'],
                             success_count=running_paths[idx]['success_count']))
 
-                    if isinstance(self.algo.replay_buffer, PathBuffer):
-                        obs_space = self._env_spec.observation_space
+                    path_dict = {}
 
-                        flattened_obses = obs_space.flatten_n(
+                    if isinstance(self.algo.replay_buffer, HerReplayBuffer):
```

Comment:
This seems kinda hacky (even if it's efficient). Can you just unflatten the observations in `HerReplayBuffer` using the `EnvSpec`? This way we can more easily delete `OffPolicyVectorizedSampler` (instead of adding more logic to it).